modules/op-running-unpriveleged-builds-in-a-container.adoc

```diff
@@ -0,0 +1,62 @@
+:_content-type: PROCEDURE
+
+[id='op-running-unpriveleged-builds-in-a-container._{context}']
+= Running unprivileged builds in a container
+
+You can run builds as a non-root user inside a container to restrict specific container activities and enhance security.
+
+[INFO]
+====
+To run as a non-root user inside the container, your container must first add the `subuid` and `subgid` with the latest shadow-utils.
+====
+
+.Procedure
+
+. Use the public https://catalog.redhat.com/software/containers/ubi8/buildah/602686f7b16b1eb2e30807ee?container-tabs=dockerfile[Containerfile] image based on Red Hat UBI.
+
+. Force run the container as the “build” user to prevent it from being “randomed” which will not have the setup needed to run as non-root.
+
+. Add `securityContext`` to each build and push steps to directly edit or export to a specific namespace the buildah ClusterTask.
++
+[source,yaml,subs="attributes+"]
+----
+securityContext:
+    runAsUser: 1000
+----
+
+. Change the `mountPath` for the build to your local user container directory to use the image cache.
++
+[source,yaml,subs="attributes+"]
+----
+volumeMounts:
+    - name: varlibcontainers
+      mountPath: /home/build/.local/share/containers
+----
+
+. Test it using https://github.com/tektoncd/pipeline/blob/main/docs/workspaces.md#configmap[ConfigMap workspace] along with a sample DockerFile and a sample taskrun.
++
+[source,terminal,subs="attributes+"]
+----
+$ tkn tr logs -Lf buildah-run
+----
++
+The following output is displayed:
++
+[source,terminal]
```

Comment:
```suggestion
+
.Example output
[source,terminal]
```

Reply:
Please replace  `<Harsh, describe what the user should see>`  with a description.

Reply:
Need SME input here.